frontend/search/searchController.js

```diff
@@ -72,16 +81,17 @@
             var actuationDefaultValue = searchCtrl.searchActuation === "Pesquisar em todas as áreas";
             var stateDefaultValue = searchCtrl.searchState === "Pesquisar em todos os estados";
 
+            //That's necessary once the function needs to know what is the attribute selected
             var natureIsNotSelected = !searchCtrl.searchNature || natureDefaultValue;
             var actuationIsNotSelected = !searchCtrl.searchActuation || actuationDefaultValue;
             var stateIsNotSelected = !searchCtrl.searchState || stateDefaultValue;
 
-            // True If the fields are the same or If the field is not selected. 
+            // True If the institution's fields and controller's fields are the same or If the field is not selected. 
             var sameNature = legalNatures[institution.legal_nature] === searchCtrl.searchNature || natureIsNotSelected;
             var sameActuationArea = actuationAreas[institution.actuation_area] === searchCtrl.searchActuation || actuationIsNotSelected;
             var sameState = stateIsNotSelected || institution.federal_state === searchCtrl.searchState.nome;
 
-            // It keeps If the search's current attribute is the same in the institution and the search param and
+            // It stores If the search's current attribute is the same in the institution and in the search param and
```

Comment:
If you separated this function with comment blocks, that's it's a sign that you have different contexts, and that should be another function well-documented.